### PR TITLE
[Enhancement] Remove the needed for neighbor host DVR mac in L3 neighbor configuration

### PR DESCRIPTION
**Pull request by er1cthe0ne**
Using DVR mac prefix (e.g., fe:16:11:) together with internal vlan ID based on incoming tunnel ID, we can determine the incoming packet is coming from a neighbor host DVR and restore src mac to the port's gateway MAC. This removes the need of neighbor host DVR mac for L3 neighbor configuration. 

This change includes:
1. define a mac prefix to indicate it is a host dvr mac (aca_config.h)
2. update L3 neighbor configuration handling (aca_ovs_l3_programming.cpp)
3. update routing test to confirm passing with new algorithm (aca_test.cpp) 

Thanks to the suggestion from https://github.com/futurewei-cloud/alcor/issues/375

### CODE REVIEW

File: include/aca_config.h [hunk 1]
```diff
@@ -18,4 +18,9 @@
 #define MAX_PORT_SCAN_RETRY 300
 #define PORT_SCAN_SLEEP_INTERVAL 1000 // 1000ms = 1s
 
+// prefix to indicate it is an Alcor Distributed Router host mac
+// (aka host DVR mac)
+#define HOST_DVR_MAC_PREFIX "fe:16:11:"
```
Comment (xieus) [inline include/aca_config.h:23]: As discussed, we could move this DVR_MAC_PREFIX to a configuration file later. Do we have an existing configuration file as of now?

File: src/README.md [hunk 2]
```diff
@@ -4,18 +4,20 @@ Next Generation Cloud Network Control Agent
 # Summary
 Source code folder:
 
-- Comm: Library for communication with network controllers and transit daemon
+- comm: Library for communication with Alcor controllers, includes gRPC and MQ implementation
+- dhcp: DHCP server implementation
```
Comment (xieus) [inline src/README.md:8]: 👍

File: src/README.md [hunk 3]
```diff
@@ -4,18 +4,20 @@ Next Generation Cloud Network Control Agent
...
+- dp_abstraction: Data plane abstraction layer implementation
 - grpc: Auto generated source codes and library for gRPC interface with Alcor Controllers
```
Comment (xieus) [inline src/README.md:10]: Find a place for MQ?

File: src/ovs/aca_ovs_l2_programmer.cpp [hunk 4]
```diff
@@ -189,10 +189,6 @@ int ACA_OVS_L2_Programmer::configure_port(const string vpc_id, const string port
     throw std::invalid_argument("tunnel_id is 0");
   }
 
-  if (overall_rc != EXIT_SUCCESS) {
-    throw std::runtime_error("Invalid environment with br-int and br-tun");
```
Comment (xieus) [inline src/ovs/aca_ovs_l2_programmer.cpp:193]: Is this validation unused any more?

**Review by xieus — APPROVED** [review]
LGTM

File: include/aca_config.h (hunk 1 above)
Comment (er1cthe0ne) [reply include/aca_config.h:23]: We don't have an existing configuration file now, we can put everything in this file to it when we create a new one.

File: src/ovs/aca_ovs_l2_programmer.cpp (hunk 4 above)
Comment (er1cthe0ne) [reply src/ovs/aca_ovs_l2_programmer.cpp:193]: correct, it is leftover from the previous logic.

**Pull request merged**